File: examples/OmeroClients/configuration.cpp
#include <omero/client.h>
#include <iostream>

int main(int argc, char* argv[]) {

    // All configuration in file pointed to by
    // --Ice.Config=file.config
    // No username, password entered
    omero::client1(argc, argv);
    /* client1.createSession(); */

    // Most basic configuration.
    // Uses default port 4063
    // createSession needs username and password
    omero::client2("localhost");
    /* client2.createSession("root", "ome"); */

    // Configuration with port information
    omero::client3("localhost", 24063);
    /* client3.createSession("root", "ome"); */

    // Advanced configuration in C++ takes place
    // via an InitializationData instance.
    Ice::InitializationData data;
    data.properties = Ice::createProperties();
    data.properties->setProperty("omero.host", "localhost");
    omero::client client4(data);
    /* client4.createSession("root", "ome"); */

    // std::map to be added (ticket:1278)
    data.properties->setProperty("omero.user", "root");
    data.properties->setProperty("omero.pass", "ome");
    omero::client client5(data);
    // Again, no username or password needed
    // since present in the data. But they *can*
    // be overridden.
    /* client5.createSession(); */
}

File: examples/OmeroClients/configuration.java
public class configuration {

    public static void main(String[] args) throws Exception {

	// All configuration in file pointed to by
	// --Ice.Config=file.config
	// No username, password entered
	omero.client client1 = new omero.client(args);
	/* client1.createSession(); */

	// Most basic configuration.
	// Uses default port 4063
	// createSession needs username and password
	omero.client client2 = new omero.client("localhost");
	/* client2.createSession("root", "ome"); */

	// Configuration with port information
	omero.client client3 = new omero.client("localhost", 10463);
	/* client3.createSession("root", "ome"); */

	// Advanced configuration can also be done
	// via an InitializationData instance.
	Ice.InitializationData data = new Ice.InitializationData();
	data.properties = Ice.Util.createProperties();
	data.properties.setProperty("omero.host", "localhost");
	omero.client client4 = new omero.client(data);
	/* client4.createSession("root", "ome"); */

	// Or alternatively via a java.util.Map instance
	java.util.Map<String, String> map = new java.util.HashMap<String, String>();
	map.put("omero.host", "localhost");
	map.put("omero.user", "root");
	map.put("omero.pass", "ome");
	omero.client client5 = new omero.client(map);
	// Again, no username or password needed
	// since present in the map. But they *can*
	// be overridden.
	/* client5.createSession(); */
    }

}
